core/apiCalls.js

```diff
@@ -66,11 +66,47 @@ export function fetchBlueprintInfoApi(blueprintName) {
 
 export function fetchModalCreateImageTypesApi() {
   const imageTypes = utils.apiFetch(constants.get_image_types)
-    .then(data => data.types)
+    .then(data => data.types.map(type => {
+      type.label = mapImageTypeLabels(type.name);
+      return type;
+    }))
     .catch(e => console.log('Error getting component types', e));
   return imageTypes;
 }
 
+export function mapImageTypeLabels(name) {
+  let label = '';
+  switch (name) {
+    case 'ami':
+      label = 'Amazon Machine Image Disk (.ami)';
+      break;
```

Comment:
I prefer using a dictionary, because that's more conventional in JavaScript than `switch`. Also, it avoids the extra function. Something like this at the top of the file:

```js
const imageTypeLabels = {
  "ami": "Amazon Machine Image Disk (.ami)",
  // ...
}
```

Reply:
We still need to pass through the image type directly, in the case where no matching label is found.  So this may not avoid an extra function.